### PR TITLE
Updates for Cloud Pak for Data 2.5

### PR DESCRIPTION
**Pull request by scottdangelo**
Updates to README for new UI in CPD 2.5
Updates to README for new testing.
Updates to Notebook.
New images for UI.
New TOKEN info for flaskapp and env.sample

Add missing flaskapp/ directory and contents.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -102,17 +103,21 @@ When the Jupyter notebook is loaded and the kernel is ready then we can start ex
 
 ![Notebook loaded](doc/source/images/wml-3-notebook-loaded.png)
 
+> **Important**: *Make sure that you stop the kernel of your notebook(s) when you are done, in order to prevent leaking of memory resources!*
```
Comment (markstur) [inline README.md]: this sounds like a "memory leak" bug.  I'm hoping you meant account quota resource in general and in that case I'd suggest you change "prevent leaking..." to something like "conserve resources."

File: README.md [hunk 2]
```diff
@@ -147,95 +152,137 @@ MODEL_NAME = "user123 customer churn model"
...
+## 6. Create a Space for Machine Learning Deployments
 
-![Commit the project changes](doc/source/images/project-1-git-commit.png)
+Go the (☰) menu and click `Analyze` -> `Analytics deployments`:
 
-A list of the assets will appear that were created in this project. Provide a commit message to identify the changes being pushed.
+![(☰) Menu -> Analytics deployments](doc/source/images/ChooseAnalyticsDeployments.png)
 
-![Provide a commit message](doc/source/images/project-2-git-commit-message.png)
+Click on `+ New deployment space):
```
Comment (markstur) [inline README.md]: Looks like an open back-tick on this line

File: README.md [hunk 3]
```diff
@@ -76,19 +80,16 @@ git clone https://github.com/IBM/telco-customer-churn-on-icp4d/
...
 ![Add data set](doc/source/images/cpd-add-data-set.png)
 
-Either drag the `Telco-Customer-Churn.csv` file to the window or navigate to it using `Select from your local file system`:
-
-![Select data set](doc/source/images/cpd-select-file-data.png)
-
 ## 3. Import notebook to Cloud Pak for Data
 
-At the project overview click the *New Asset* button, and choose *Add notebook*.
+At the project overview, either click the `+Add to project` button, and choose `Notebook`, or, if the *Notebooks* section exists,  to the right of *Notebooks* click `+ New notebook`:
```
Comment (rhagarty) [inline README.md:89]: add line break between 2nd and 3rd line.

**Review by markstur — CHANGES_REQUESTED** [review]
a couple little things...

File: README.md [hunk 4]
```diff
@@ -102,17 +103,21 @@ When the Jupyter notebook is loaded and the kernel is ready then we can start ex
...
 Spend a minute looking through the sections of the notebook to get an overview.
 
 In section `2.0 Load and Clean data`, highlight the cell labelled `# Place cursor ...` by clicking on it. Click on the *10/01* button to select a specific data set.
 
-If you are following the [Learning Path](), Choose The *Remote* tab, and pick the virtualized data set that has all three joined tables (i.e. `User999.billing+products+customers`), and choose `Insert Pandas DataFrame`.
+If you are following the [CP4D Learning Path](https://developer.ibm.com/series/cloud-pak-for-data-learning-path/), Choose The *Files* tab, and pick the virtualized data set that has all three joined tables (i.e. `User<xyz>.BILLINGPRODUCTSCUSTOMERS`), and choose `Insert Pandas DataFrame`.
 
 ![Add remote Pandas DataFrame](doc/source/images/wml-4-add-dataframe.png)
 
-If you are running this pattern alone, choose the *Local* tab, and choose the 'Telco-Customer-Churn.csv` file that you added, then choose `Insert Pandas DataFrame`:
```
Comment (rhagarty) [inline README.md:115]: this is confusing. recommend sticking with your blanket statement "if you are following ..."

File: README.md (hunk 2 above)
Comment (rhagarty) [inline README.md]: missing a tic mark

File: README.md [hunk 5]
```diff
@@ -102,17 +103,21 @@ When the Jupyter notebook is loaded and the kernel is ready then we can start ex
...
+If you are running this pattern alone, choose the *Files* tab, and choose the 'Telco-Customer-Churn.csv` file that you added, then choose `Insert Pandas DataFrame`:
```
Comment (markstur) [inline README.md]: The single quote in front of Telco-Cust needs to be a back-tick.  The formatting is messed up.

File: README.md [hunk 6]
```diff
@@ -121,7 +126,7 @@ By adding data a block of code will be added to the notebook. The code will auto
 
 ![Generated code to handle Pandas DataFrame](doc/source/images/wml-5-generated-code-dataframe.png)
 
-> **IMPORTANT**: Don't forget to update the next cell `df = df1` (or whatever your dataframe number is) with the variable from the generated code, in the case of `Remote` Virtualized data. If you are using the `Local` data, use `df = df_data_1` (or whatever your dataframe number is).
+> **IMPORTANT**: Don't forget to update the next cell to assign the `df` variable. In the case of Virtualized data, it will look like `df = data_df_1`, `data_df_2, etc. For a locally uploaded file, it will look like `df = df_data_1` or `df_data_X` depending on your number `X` for the variable from the generated code.
```
Comment (markstur) [inline README.md]: missing backtick after data_df_2.  The formatting is messed up.

File: README.md (hunk 1 above)
Comment (scottdangelo) [reply README.md]: Done.

File: README.md (hunk 2 above)
Comment (scottdangelo) [reply README.md]: done

File: README.md (hunk 5 above)
Comment (scottdangelo) [reply README.md]: done

File: README.md (hunk 6 above)
Comment (scottdangelo) [reply README.md]: done

**Review by rhagarty — CHANGES_REQUESTED** [review]
- check for missing links (bracketed text with no associated URL)
- `Learn More` section bullets should all end in period
- recommend changing line: (also contains a broken link)
If you are not on the [ICP4D Learning Path], which uses Virtualized Data, upload the dataset into your project now.
:TO:
If you are on the ICP4D learning path, which uses Virtualized Data, you can skip this section.

File: README.md [hunk 7]
```diff
@@ -147,95 +152,137 @@ MODEL_NAME = "user123 customer churn model"
...
+curl -k -X POST --header 'Content-Type: application/json' --header 'Accept: application/json' --header "Authorization: Bearer  $WML_AUTH_TOKEN" -d '{"input_data": [{"fields": ["gender","SeniorCitizen","Partner","Dependents","tenure","PhoneService","MultipleLines","InternetService","OnlineSecurity","OnlineBackup","DeviceProtection","TechSupport","StreamingTV","StreamingMovies","Contract","PaperlessBilling","PaymentMethod","MonthlyCharges","TotalCharges"],"values": [["Female",0,"No","No",1,"No","No phone service","DSL","No","No","No","No","No","No","Month-to-month","No","Bank transfer (automatic)",25.25,25.25]]}]}' $URL
 ```
 
+A json string will be returned with the response, including a "Yes" of "No" at the end indicating the prediction of if the customer will churn or not.
+
 ## 9. Create a Python Flask app that uses the model
```
Comment (markstur) [inline README.md:287]: shouldn't say "workshop" in a code pattern

**Review by markstur — COMMENTED** [review]
OK got to the bottom. See comments and look for unfinished linking in the readme.

File: README.md (hunk 3 above)
Comment (scottdangelo) [reply README.md:89]: done

File: README.md (hunk 4 above)
Comment (markstur) [reply README.md:115]: I actually thought using both example was overkill when one should work, but was going to leave that up to you. If you can simplify it. Simple is better.  Add a file called "this" or "that". -- but you probably have your reasons.

Comment (scottdangelo) [reply README.md:115]: Changed to "If you are not following the CPD Learning Path and instead using the code pattern by itself, ..."

File: README.md (hunk 2 above)
Comment (scottdangelo) [reply README.md]: done

File: README.md (hunk 7 above)
Comment (scottdangelo) [reply README.md:287]: done.

**Review by rhagarty — APPROVED** [review]

**Review by markstur — APPROVED** [review]

**Pull request merged**